pkg/standalone-utils/contracts/relayer/BaseRelayerLibrary.sol

```diff
@@ -138,25 +151,33 @@ contract BaseRelayerLibrary is IBaseRelayerLibrary {
     }
 
     /**
-     * @dev Returns the amount referenced by chained reference `ref`. Reading an amount clears it, so they can each
-     * only be read once.
+     * @dev Returns the amount referenced by chained reference `ref`.
+     * If the reference is temporary, it will be cleared after reading it, so they can each only be read once.
+     * If the reference is not temporary (i.e. read-only), it will not be cleared after reading it
+     * (see `_isTemporaryChainedReference` function).
      */
     function _getChainedReferenceValue(uint256 ref) internal override returns (uint256 value) {
-        bytes32 slot = _getTempStorageSlot(ref);
+        bytes32 slot = _getStorageSlot(ref);
```

Comment:
Note: in #1507 the suggestion was to return a flag in `_getTempStorageSlot`, but the method is also used in `_setChainedReferenceValue`, and we don't need that flag there.